src/solver/outer/forward_outer.cpp

```diff
@@ -0,0 +1,227 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/outer/solvers.hpp>
+
+#include <miopen/outer/invoke_params.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/outer.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/target_properties.hpp>
+
+static const size_t LOCAL_SIZE = 256;
+
+namespace miopen {
+
+namespace solver {
+
+namespace outer {
+
+static bool IsImprovementOverROCm(const miopen::outer::ProblemDescription& problem)
+{
+    auto dtype = problem.GetX1Desc().GetType();
+    auto ydims = problem.GetYDesc().GetLengths();
+
+    if(dtype == miopenHalf)
+    {
+        if(ydims[0] <= 512)
+        {
+            return true;
+        }
+        else if(ydims[0] <= 2048)
```

Comment:
Better to leave `return false;` at the end of `IsImprovementOverROCm` and keep the dim conditions in single lines.
```suggestion
        else if(ydims[0] <= 2048 && ydims[1] <= 2048)
```

Reply:
I leaved out "return false" at the end of "IsImprovmentOverROCm" functions. In addition, I replaced hierarchical conditions with singe line conditions.

Reply:
Maybe I commented too short to be delivering my intention. I meant **single lines** and leaving `return false` at the end like this.

```
static bool IsImprovementOverROCm(const miopen::outer::ProblemDescription& problem)
{
    auto dtype = problem.GetX1Desc().GetType();
    auto ydims = problem.GetYDesc().GetLengths();

    if(ydims[0] <= 512)
    {
        return true;
    }
    else if(2048 < ydims[0] && ydims[1] <= 128)
    {
        return true;
    }
    else if(dtype == miopenHalf || dtype == miopenBFloat16)
    {
        if(ydims[0] <= 2048 && ydims[1] <= 2048)
        {
            return true;
        }
    }
    else if(dtype == miopenFloat)
    {
        if(ydims[0] <= 2048 && ydims[1] <= 512)
        {
            return true;
        }
    }

    return false;
}
```

Reply:
It's trivial so, optional fix. Fix it if you want to.

Reply:
@kyeonghwanryu 
Actually, I tried your suggestion before, but there was a problem. If there are too many conditions with the same action (e.g., if(A) return true; else if(B) return true; else if(C) return true; ...), it doesn't pass the "make tidy" check. The "make tidy" tool recommends merging conditions (e.g., if (A || B || C) return true;) since fewer repetitions help prevent errors. Therefore, I think we have to accept the current code.

Reply:
Oh okay.